src/sentry/rules/conditions/tagged_event.py

```diff
@@ -40,12 +42,23 @@ class TaggedEventForm(forms.Form):
     def clean(self):
         super(TaggedEventForm, self).clean()
 
+        key = self.cleaned_data.get("key")
         match = self.cleaned_data.get("match")
         value = self.cleaned_data.get("value")
 
         if match not in (MatchType.IS_SET, MatchType.NOT_SET) and not value:
             raise forms.ValidationError("This field is required.")
 
+        if (
+            key == "release"
+            and value == "latest"
```

Comment:
Maybe check `value` here only if `match` is a condition that needs `value`, we actually probably don't want to clear out form values when changing options for nicer experience

Reply:
dang good catch with this, thanks for testing